### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by rwblickhan**
I added GoogleTest as a submodule and added a sample class and unit test for that class. Additionally I updated the Travis config so that it'll run all unit tests after building.

### CODE REVIEW

**Comment by bobheadxi** [comment]
I guess nothing is set in stone yet but I'm not sure on the `src/...` directory structure - I quite like how @CoffeeTableEspresso has his [YASL project set up](https://github.com/CoffeeTableEspresso/YASL) with `./compiler/`,  `./interpreter/`, etc

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "googletest"]
+	path = googletest
+	url = https://github.com/google/googletest.git
```
Comment (bobheadxi) [inline .gitmodules:3]: This generates a bunch of `CMake` stuff and git will keep telling you your working dir is dirty - adding `ignore = dirty` to this configuration will silence them

File: .travis.yml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 language: cpp
-script: mkdir build && cd build && cmake .. && make
+script: mkdir build && cd build && cmake .. && make && ./durian_test
```
Comment (bobheadxi) [inline .travis.yml]: I think `cmake`, `make`, and running `./durian_test` at least should be on separate script steps to make Travis logs easier to read and make it easier to find which step failed

**Review by bobheadxi — COMMENTED** [review]
Looks great, just a few comments. I think we'll need @CoffeeTableEspresso to enable Travis builds as well.

File: .travis.yml (hunk 2 above)
Comment (bobheadxi) [reply .travis.yml]: I also believe you are missing `git submodule update --init --recursive`

Comment (bobheadxi) [reply .travis.yml]: I've added a commit with an updated config @ 1888f2f

**Comment by bobheadxi** [comment]
Another note - it is far easier to manage our branches and work together if we just use branches within this repository. We all have push access, and the master branch should be protected - there is no need for forks.

Thoughts, @CoffeeTableEspresso @leesw98 ?

**Comment by bwdmonkey** [comment]
I agree with the notion that we should just have branches on `origin` unless there is some reason we should be using forks instead. I think it's gonna be tricky if we run into cases when we have to fix each other's PR directly (or indirectly through a PR to a PR) as you'd need to add the collaborator to our forked repos.

**Comment by rwblickhan** [comment]
Oh, I just forked it for now because Travis isn’t enabled on the main repo (and I don’t have access to enable it) but I wanted to go ahead and get the config set up. I wasn’t planning to work off the fork after setup was done.

File: .gitmodules (hunk 1 above)
Comment (rwblickhan) [reply .gitmodules:3]: Will change tonight

**Comment by bobheadxi** [comment]
Have you tried using a more modern C++ function like unique_ptr on travis? I don’t believe it works by default. Found a workaround last night for clang (should be fine with gcc though) - I’ll add a commit to this PR later tonight, and then it should be good to merge

edit: see 7e74282

**Review by bobheadxi — APPROVED** [review]
Good to go now, other than `ignore = dirty`, but that's non-blocking

**Pull request merged**